### ISSUE TITLE
macOS Compile Error: the type of this value must be known in this context

### ISSUE THREAD

**Issue by PSeitz** [ggez/ggez#285]
I can't compile on macOS, tried with the latest rust nightly.

```
error[E0619]: the type of this value must be known in this context
  --> /Users/d056831/.cargo/registry/src/github.com-1ecc6299db9ec823/app_dirs-1.1.1/src/imp/platform/macos.rs:10:40
   |
10 |         Ok(Component::RootDir.as_ref().into())
   |                                        ^^^^

```

**Comment by PSeitz** [comment]
Seems to be this issue:
https://github.com/AndyBarron/app-dirs-rs/issues/28

**Comment by icefoxen** [comment]
Yeah not super thrilled.  The maintainer did say in that issue that they were super busy, which I can sympathize with, but it'd be nice to at least have simple fixes merged...

Not sure what we can do besides fork and update it, or vendor it and add some patches...

**Comment by PSeitz** [comment]
Maybe we should temporarily redirect to the fix ```app_dirs = { git = "https://github.com/koute/app-dirs-rs" }``` and then wait what happens.

**Comment by icefoxen** [comment]
That's not a fix that will work in a released package, since you can't publish things to cargo if they depend on a git repo.

If you're using the devel branch though, that's good enough.  Will work for the moment.

**Comment by PSeitz** [comment]
Ah okay, well creating a new crate would be kinda overkill, but maybe the only solution...

**Comment by icefoxen** [comment]
Well this is also likely a bug in nightly, given that it works on all other platforms and releases.  Hopefully that'll get fixed eventually.

**Comment by icefoxen** [comment]
Fixed in PR #287, leaving this open so we keep track of it.

**Comment by icefoxen** [comment]
Fixed in release 0.4.2
